plugins/metrics/pkg/agent/node.go

```diff
@@ -208,13 +208,7 @@ func (m *MetricsNode) Discover(ctx context.Context, request *remoteread.Discover
 		return nil, fmt.Errorf("could not get driver: %w", err)
 	}
 
-	promDriver, ok := driver.(*drivers.ExternalPromOperatorDriver)
-	if !ok {
-		m.logger.Errorf("found wrong driver type '%T'", driver)
-		return nil, fmt.Errorf("could not discover Prometheus instances")
-	}
-
-	entries, err := promDriver.Discover(*request.Namespace)
+	entries, err := driver.DiscoverPrometheuses(ctx, *request.Namespace)
```

Comment:
The driver here still needs to be dependency-injected. I suggest handling it somewhere in [here](https://github.com/rancher/opni/blob/main/plugins/metrics/pkg/agent/plugin.go#L34-L54) maybe by passing the driver to NewMetricsNode.